contrib/go/src/python/pants/contrib/go/subsystems/go_distribution.py

```diff
@@ -9,54 +9,28 @@
 from collections import OrderedDict, namedtuple
 
 from pants.base.workunit import WorkUnit, WorkUnitLabel
-from pants.binaries.binary_util import BinaryUtil
+from pants.binaries.binary_tool import NativeTool
 from pants.fs.archive import TGZ
-from pants.subsystem.subsystem import Subsystem
 from pants.util.contextutil import temporary_dir
 from pants.util.memo import memoized_property
 from pants.util.process_handler import subprocess
 
 
-class GoDistribution(object):
+class GoDistribution(NativeTool):
   """Represents a self-bootstrapping Go distribution."""
 
-  class Factory(Subsystem):
-    options_scope = 'go-distribution'
+  options_scope = 'go-distribution'
+  name = 'go'
+  default_version = '1.8.3'
+  archive_type = 'tgz'
```

Comment:
Looks like maybe this bit is naive in the face of `.tgz` vs `.tar.gz`: https://github.com/pantsbuild/pants/blob/master/src/python/pants/binaries/binary_util.py#L180

Reply:
I'm just delegating to our underlying archive functionality, which assumes that `tgz` maps to `.tar.gz`. Changing that, if we perceive a need, is probably out of scope here. I mean, we know that it's `go.tar.gz`, so this works in practice.

Reply:
I should have linked the CI test failure - this was looking like a bug in the underlying extraction process.

Reply:
Ah, did not read the backtrace closely. The extraction blowup is in GoDistribution code using TGZ directly.

Reply:
Yeah, took me a moment to figure it out, but the extension thing doesn't matter - it's just how we generate a name for the dir so it doesn't collide with the name of the downloaded file. The issue was I forgot to remove the unpacking code in GoDistribution, since BinaryTool now does this for us...